provider/win_hlk_suite.py

```diff
@@ -0,0 +1,587 @@
+                zip_path = o.splitlines()[-1]
+                logging.info('Uploading the test result from %s to %s:',
+                             zip_path, host_path)
+                self._vm.copy_files_from(zip_path, host_path)
+            results = results + output
+
+        logging.info(results)
+        return results
+
+    def run_tests(self, tests_id, target_key, project_name,
+                  machine_name, pool_name, timeout=600, step=3):
+        """
+        Run tests.
+
+        :param tests_id: List of tests ID.
+        :type tests_id: list
+        :param target_key: Target key.
+        :type target_key: str
+        :param project_name: Project name.
+        :type project_name: str
+        :param machine_name: Machine name.
+        :type machine_name: str
+        :param pool_name: Pool name.
+        :type pool_name: str
+        :param timeout: Timeout for running in seconds.
+        :type timeout: int
+        :param step: Time to sleep between attempts in seconds.
+        :type step: int
+        :raise: HLKRunError, if run timeout or found error messages.
+        """
+        for test_id in tests_id:
+            self.queue_test(test_id, target_key, project_name,
+                            machine_name, pool_name)
+
+        if not utils_misc.wait_for(
+                lambda: "NotRunning" == self.get_project(project_name)['status'],
+                timeout, step=step):
+            raise HLKRunError('Timeout for running tests.')
+
+        resutls = self.list_tests_results(tests_id, target_key, project_name,
+                                          machine_name, pool_name)
+        err_msg = []
+        for result in resutls.splitlines():
+            if "Task error message" in result:
+                err_msg.append(result.strip())
+        if err_msg:
+            raise HLKRunError('Found task error messages:%s' % err_msg)
+
+    def simple_run_test(self, pool_name, project_name, target_name,
+                        tests_name, timeout=14400, step=600):
+        """
+        Simple run test.
+
+        :param pool_name: Pool name.
+        :type pool_name: str
+        :param project_name: Project name.
+        :type project_name: str
+        :param target_name: Target name.
+        :type target_name: str
+        :param tests_name: List of tests name.
+        :type tests_name: str
+        :param timeout: Timeout for running in seconds.
+        :type timeout: int
+        :param step: Time to sleep between attempts in seconds.
+        :type step: int
+        """
+        default_pool = self.get_default_pool()[0]
+        machine_name = default_pool['machine_name']
+        self.create_pool(pool_name)
+        self.move_machine_from_default_pool(machine_name, pool_name)
+        self.set_machine_state(machine_name, pool_name, STATE_READY)
+        # FIXME: Sleep to provide buffer to execute next steps.
+        time.sleep(60)
```

Comment:
On some old hosts, 30 is not enough, extend to 60.

Reply:
Thanks, have updated.